src/core.ts

```diff
@@ -31,76 +24,53 @@ import {
   OptionallyHasInitPreContext,
   OptionallyHasSanitizeBody,
   OptionallyHasSanitizeParams,
+  PreAuthReqsSatisfied,
   PromiseOrSync,
   PromiseResolveOrSync,
+  RespondReqsSatisfied,
+  SanitizeResponseReqsSatisfied,
 } from './types';
 
-// @todo: MINOR: consider bringing this back to life so complexity of optionality is removed from executeHipthrustable
 export function withDefaultImplementations<
-  TStrategy extends OptionallyHasInitPreContext<any, any> &
-    OptionallyHasSanitizeParams<any, any> &
-    OptionallyHasSanitizeBody<any, any> &
-    HasAllRequireds &
-    OptionallyHasAttachData<any, any> &
-    OptionallyHasDoWork<any, any>
->(
-  strategy: TStrategy
-): {
-  initPreContext: TStrategy extends HasInitPreContext<any, any>
-    ? TStrategy['initPreContext']
-    : () => {};
-  sanitizeParams: TStrategy extends HasSanitizeParams<any, any>
-    ? TStrategy['sanitizeParams']
-    : () => {};
-  sanitizeBody: TStrategy extends HasSanitizeBody<any, any>
-    ? TStrategy['sanitizeBody']
-    : () => {};
-  preAuthorize: TStrategy['preAuthorize'];
-  attachData: TStrategy extends HasAttachData<any, any>
-    ? TStrategy['attachData']
-    : () => {};
-  finalAuthorize: TStrategy['finalAuthorize'];
-  doWork: TStrategy extends HasDoWork<any, any>
-    ? TStrategy['doWork']
-    : () => {};
-  respond: TStrategy['respond'];
-  sanitizeResponse: TStrategy['sanitizeResponse'];
-} {
-  // @tswtf why isn't typescript smart enough to not require these type assertions below?
+  TStrategy extends HasAllRequireds &
+    HasAllNotRequireds &
+    PreAuthReqsSatisfied<TStrategy> &
+    AttachDataReqsSatisfiedOptional<TStrategy> &
+    FinalAuthReqsSatisfied<TStrategy> &
+    DoWorkReqsSatisfiedOptional<TStrategy> &
+    RespondReqsSatisfied<TStrategy> &
+    SanitizeResponseReqsSatisfied<TStrategy>
+>(strategy: TStrategy): HasAllStagesNotOptionals {
   return {
```

Comment:
- [ ] please order the requirements above in a logical fashion

Reply:
done